restricted_site_access.php

```diff
@@ -325,6 +326,13 @@ public static function restrict_access( $wp ) {
 				if ( ( $current_url_path === $redirect_url_without_scheme ) || ( $redirect_url_without_scheme === $current_url_without_scheme ) ) {
 					return;
 				}
+
+				$redirection_url_host = trailingslashit( wp_parse_url( $results['url'], PHP_URL_HOST ) );
+				$current_url_host     = trailingslashit( wp_parse_url( home_url( $wp->request ), PHP_URL_HOST ) );
+
+				if ( $redirection_url_host === $current_url_host || '/' === $redirection_url_host ) {
+					$_SESSION['rsa_redirect'] = 1;
```

Comment:
I'm trying to figure out how this is used, is it used across two HTTP requests?

Sessions make caching difficult but that may not be a big problem due to the nature of this plugin.